src/main/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidator.java

```diff
@@ -136,12 +138,114 @@ private void validateSubResourceTotal(IntangibleAssetsResource intangibleAssetsR
 
     }
 
-    private Long getCostAtPeriodEnd(IntangibleAssetsResource intangibleAssetsResource) {
+    private void validateAdditionsTotal(Errors errors, IntangibleAssets intangibleAssets) {
 
-        return Optional.ofNullable(intangibleAssetsResource)
-                .map(IntangibleAssetsResource::getCost)
-                .map(Cost::getAtPeriodEnd)
-                .orElse(0L);
+        Long goodwillAdditions =
+                getAdditions(intangibleAssets.getGoodwill());
+
+        Long otherAdditions =
+                getAdditions(intangibleAssets.getOtherIntangibleAssets());
+
+
+        Long resourceAdditionsTotal = goodwillAdditions + otherAdditions;
+
+        Long additions = getAdditions(intangibleAssets.getTotal());
+
+        if (!additions.equals(resourceAdditionsTotal)) {
+
+            addError(errors, incorrectTotal, getJsonPath(IntangibleSubResource.TOTAL, ADDITIONS));
+        }
+    }
+
+    private void validateDisposalsTotal(Errors errors, IntangibleAssets intangibleAssets) {
+
+        Long goodwillDisposals =
+                getDisposals(intangibleAssets.getGoodwill());
+
+        Long otherDisposals =
+                getDisposals(intangibleAssets.getOtherIntangibleAssets());
+
+        Long resourceDisposalsTotal = goodwillDisposals + otherDisposals;
+
+        Long disposals = getDisposals(intangibleAssets.getTotal());
+
+        if (!disposals.equals(resourceDisposalsTotal)) {
+            addError(errors, incorrectTotal, getJsonPath(IntangibleSubResource.TOTAL, ADDITIONS));
```

Comment:
The json path should be relevant to the field; i.e. `DISPOSALS`, rather than `ADDITIONS` here